src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -1481,44 +1272,44 @@ class GraphColorDistance2
                 // * The Distance-1 code used the knowledge of the degree of the vertex to cap the number of iterations
                 //   but in distance-2 we'd need the total vertices at distance-2 which we don't easily have aprioi.
                 //   This could be as big as all the vertices in the graph if diameter(G)=2...
-                for(color_type offset = 0; !foundColor && offset < nv; offset += VB_D2_COLORING_FORBIDDEN_SIZE)
+                for(color_type offset = 1; offset <= nv; offset += VB_D2_COLORING_FORBIDDEN_SIZE)
```

Comment:
We don't stop immediately after finding colors now ? May be I am missing a break somewhere.

Reply:
@srajama1 Down below when we actually color the vertex, the functor just returns immediately. That way we don't need nested breaks.